include/emp/web/Color.hpp

```diff
@@ -0,0 +1,496 @@
+        if (str.length() && str.back() == '%') {
+            return clamp_css_float(parseFloat(str) / 100.0f);
+        } else {
+            return clamp_css_float(parseFloat(str));
+        }
+    }
+    /// Convert floats from hsl to rgb value
+    /// @param m1 Value calculated in ParseHSl
+    /// @param m2 Value calculated in ParseHSl
+    /// @param h Value calculated in ParseHSl
+    float css_hue_to_rgb(const float m1, const float m2, float h) {
+        if (h < 0.0f) {
+            h += 1.0f;
+        } else if (h > 1.0f) {
+            h -= 1.0f;
+        } else if (h * 6.0f < 1.0f) {
+            return m1 + (m2 - m1) * h * 6.0f;
+        } else if (h * 2.0f < 1.0f) {
+            return m2;
+        }
+        else if (h * 3.0f < 2.0f) {
+            return m1 + (m2 - m1) * (2.0f / 3.0f - h) * 6.0f;
+        }
+        return m1;
+    }
+    /// Split a string into a vector of strings
+    /// @param s The string to split
+    /// @param delim The delimiter
+    emp::vector<std::string> split(const std::string& s, char delim) {
+        emp::vector<std::string> elems;
+        std::stringstream ss(s);
+        std::string item;
+        while (std::getline(ss, item, delim)) {
+            elems.push_back(item);
+        }
+        return elems;
+    }
+
+  }
+
+  /// Color constructor
+  /// @param css_str The string to construct a color from
+  Color::Color( const std::string& css_str ) {
+   std::string str = css_str;
+
+    // Remove all whitespace, not compliant, but should just be more accepting.
+    emp::remove_whitespace(str);
+
+    // Convert to lowercase.
+    emp::to_lower(str);
+
+    // Ensure sorted and find named color match with lower_bound
+    emp_assert(std::is_sorted(
+        std::begin(color_impl::namedColors), std::end(color_impl::namedColors), 
+        [](const auto& left, const auto& right){
+            return left.name < right.name;
+        }));
+    
+    auto namedColor = std::lower_bound(
+        std::begin(color_impl::namedColors), std::end(color_impl::namedColors), str,
+        [](const auto& left, const auto& right){
+            return left.name < right;
+        });
+
+    if (namedColor->name == str){
+        *this = namedColor->color;
+        (*this).cached_css_str = emp::ColorRGB(r, g, b, a);
+        //(*this).cached_sf_color = (sf::Color)*this;
+        return;
+    }
+
+    *this = ParseColor(str);
+
+  }
+  
+    /// Handle all color detection and parsing from the string
+    /// @param str The color string to be parsed
+    Color Color::ParseColor(const std::string& str) {
+        // #abc and #abc123 syntax.
+        if (DetectSyntaxABC(str)) {
+            return ParseABC(str);
+        }
+
+        size_t open_paren = str.find_first_of( '(' );
+        if (open_paren != std::string::npos) {
+            const std::string fname = str.substr(0, open_paren);
+
+            if (fname == "rgba" || fname == "rgb") {
```

Comment:
can rename fname to "format_name" throughout? I keep seeing it everywhere and thinking "filename"